### PR TITLE
fix signature validation

### LINKED ISSUE #25: middleware skips signature validation when 'x-line-signature' header is not set.
in [middleware.ts](https://github.com/line/line-bot-sdk-nodejs/blob/master/lib/middleware.ts#L24)
```typescript
    const signature = req.headers["x-line-signature"] as string;

    if (!signature) {
      next();
      return;
    }
```

what happens if someone request without `x-line-signature` header?

### PR DESCRIPTION
**Pull request by musou1500**
Resolve #25

### CODE REVIEW

**Comment by musou1500** [comment]
I signed the [ICLA](https://feedback.line.me/enquete/public/919-h9Yqmr1u).

File: lib/middleware.ts [hunk 1]
```diff
@@ -22,7 +22,7 @@ export default function middleware(config: Line.Config & Line.MiddlewareConfig):
     const signature = req.headers["x-line-signature"] as string;
 
     if (!signature) {
-      next();
+      next(new SignatureValidationFailed("no signature", signature));
```
Comment (noraesae) [inline lib/middleware.ts]: `signature` here may be `undefined`, so it is not really meaningful to pass it. How about making `public signature` in `SignatureValidationFailed` optional and just not passing it?

- https://github.com/line/line-bot-sdk-nodejs/blob/master/lib/exceptions.ts#L4

``` typescript
constructor(
  message: string,
  public signature?: string,
) {
```

**Comment by musou1500** [comment]
@noraesae thank you for your review. done.

**Comment by noraesae** [comment]
Thanks. This change will be published with the next version, which is not yet specifically scheduled.

**Pull request merged**